src/devices/src/virtio/persist.rs

```diff
@@ -97,6 +121,30 @@ impl VirtioDeviceState {
             activated: device.is_activated(),
         }
     }
+
+    /// Does a sanity check against expected values.
+    pub fn sanity_check(
+        &self,
+        device_type: u32,
+        num_queues: usize,
+        queue_max_size: u16,
+    ) -> std::result::Result<(), Error> {
+        // Check:
+        // - right device type,
+        // - acked features is a subset of available ones,
+        // - right number of queues,
+        if self.device_type != device_type
+            || (self.acked_features & !self.avail_features) != 0
```

Comment:
nit: use xor for this evaluation.

`self.acked_features ^ self.avail_features != 0`

Reply:
actually no, while `a & !b` implies `a ^ b` the reverse is not necessarily true.
`a ^ b` = `(a & !b) | (!a &b)` so using `xor` could evaluate true to because of `(!a &b)` which would provide unwanted behavior in our case.

Practical example:
```
acked = 0b0001
avail = 0b0101
# above is correct config example
acked & !avail = 0 # checking against 0 confirms config is good -> correct
acked ^ avail = 0b0100 # checking against 0 says the config isn't good -> incorrect
```

Reply:
I see, I completely forgot that the device does not necessarily need to accept all available features, therefore the XOR becomes incorrect.